### ISSUE TITLE
0x9554bD0533Ec4828cBfEa0476Acd62b90B0692D9 not detected as honeypot (but it is)

### ISSUE THREAD

**Issue by phackt** [malvaphe/Crypto-Honeypot-Detector#5]
Hi Nicolò, it could be great to investigate why this token is not detected as a honeypot:
`{"data":{"isHoneypot":false,"buyFee":"4.0","sellFee":"4.0","buyGas":"116252","sellGas":"79093","maxTokenTransaction":null,"maxTokenTransactionMain":null,"tokenSymbol":"ORE","mainTokenSymbol":"USDT","priceImpact":"1.3","problem":false,"extra":null}}`

However, testing with [honeypot.is](https://honeypot.is/?address=0x9554bD0533Ec4828cBfEa0476Acd62b90B0692D9):
`HONEYPOT DETECTED: EXECUTION REVERTED: TRANSFERHELPER: TRANSFER_FROM_FAILED`

If you have any idea. Thanks mate.

**Comment by malvaphe** [comment]
Hi @phackt ,
today I tried to do a test, it's been a while since I updated and checked this project, the result is the following:
{"data":{"isHoneypot":false,"buyFee":"4.0","sellFee":"3.9","buyGas":"116158","sellGas":"78999","maxTokenTransaction":null,"maxTokenTransactionMain":null,"tokenSymbol":"ORE","mainTokenSymbol":"USDT","priceImpact":"3.2","problem":true,"extra":"Price change after the swaps is 3.2%, which is really high! (Too high percentages can cause false positives)"}}

I tried running the test in honeypot.is as well, the result (HONEYPOT DETECTED: EXECUTION REVERTED: TRANSFERHELPER: TRANSFER_FROM_FAILED), from what I can see, refers to a purchase problem, honeypot.is is unable to purchase the token.

So I tried to perform the various steps by hand, using PancakeSwap. In order to buy the token, I had to set the Slippage Tolerance to 40%. Now the part that interests us, will I be able to sell it? The answer is... no, but the problem is another: "Unexpected issue with estimating the gas. Please try again." (usually suggests that we haven't approved the token sale, or there's a problem with the token's approve function).

There remains only one last test to check whether it is an error or not, try to actually perform the action of the honeypot detector, i.e. the simulation becomes reality. And the result is: you can sell (it would seem possible only using the contract of the honeypot detector...).
![tx](https://user-images.githubusercontent.com/49184710/236774391-4f93e796-dd62-4974-8120-fb53ff81cc4e.jpg)

As seen from the transaction, the contract converts some WBNB to USDT, then USDT to ORE, and then a portion of ORE back to USDT (even if it signals us an internal error in the transaction: "Although one or more Error Occurred [execution reverted] Contract Execution Completed").

So, in conclusion, it's like a sort of "front end honeypot". Thinking about it quickly, it should be a problem with the approve function. To better understand what the problem is, you should analyze the contract, but right now I don't have too much time... it would be a fun thing to do.

Thanks for the report,
maybe someone who wants to analyze the contract will find the answer in the future.
Have a nice day :)

**Comment by phackt** [comment]
Thanks a lot it's really interesting, also i ll try to dive into it when i ll have a bit of time to understand what's the contract is doing here. 
Have a nice day too !
